hash_practice/exercises.py

```diff
@@ -2,18 +2,41 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: n^2logn

```

Comment:
I get what you're going for here and it makes sense, but it seems you're counting `n` as both the number of words in the list and the number letters in the word, and you can't really use the same variable twice like that. Let's call the number of letters in a word `m` and then we would get  O(n*m*log(m)), which is my preferred answer.

However we are also just accepting O(n) because while the number of letters in a word has a definite effect on the run time, English words are pretty limited in how long they get (only 5 letters on average). This means that 'm' is going to be dwarfed by 'n' to the point that we can ignore `m`. After all there can easily be hundreds or thousands of words in the list, but there are unlikely to be hundreds of letters in a word.